services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTestImpl.scala

```diff
@@ -0,0 +1,49 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dao.repository.ProjectConfigurationRepositoryTestImpl
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class ProjectConfigurationServiceTestImpl(
+  repositoryTestImpl: ProjectConfigurationRepositoryTestImpl,
+  testMode: ServiceTestMode
+)(
+  implicit ec: ExecutionContext
+) extends ProjectConfigurationService {
+
+  override def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] =
+    testMode match {
+      case WithException => Future.failed(new RuntimeException("Something went wrong"))
+      case _             => Future.unit
+    }
+
+  override def getLastByProjectId(projectId: ProjectId, userId: UserId): Future[Option[ProjectConfiguration]] =
+    testMode match {
+      case WithReturnNone => Future.successful(None)
+      case _              => repositoryTestImpl.getAllByProjectId(projectId).map(_.sortBy(_.version).lastOption)
```

Comment:
```suggestion
      case _              => repositoryTestImpl.getAllByProjectId(projectId).map(_.lastOption)
```
Keep it as simple as possible